File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int dp[46];
    int climbstairs(int n) {
        if(n==0 ) return 1;
        if(n<0) return 0;
        
        if(dp[n]!=-1) return dp[n];
        return dp[n]=climbstairs(n-1)+climbstairs(n-2);
    }
    int climbStairs(int n) {
       
        memset(dp,-1,sizeof(dp));
        return climbstairs(n);
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n==2 :
                return 2
        if n==1 :
                return 1
        
        
        
        
       
        
        
        
        
        a=1
        prev=2
    
        for i in range(3, n+1):
                  
            curr=a+prev;
            a=prev;
            prev=curr;
        
        
        return prev
        
     
        
        
     
        